src/ui/components/Collections/CollectionRow.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useReducer, useState } from 'react';
+import { TableCell, TableRow } from '@material-ui/core';
+import PropTypes from 'prop-types';
+import CreateIcon from '@material-ui/icons/Create';
+import ReactJson from 'react-json-view';
+import useStyles from './styles';
+import SaveIcon from '@material-ui/icons/Save';
+import CancelIcon from '@material-ui/icons/Cancel';
+import axios from 'axios';
+import { useQueryClient } from 'react-query';
+import AlertDialog from './AlertDialog';
+
+function CollectionRow(props) {
+  const classes = useStyles();
+  const queryClient = useQueryClient();
+
+  const reducer = (state, action) => {
+    return { ...state, [action.header]: action.value };
+  };
+  const { headers, data, selectedCollection, editable, addNew, callback } = props;
+  const [edit, setEdit] = useState(addNew);
+  const [state, dispatch] = useReducer(reducer, data);
+
+  const updateData = () => {
+    const bundle = {};
+    headers.forEach(header => {
+      if (!header.viewOnly) {
+        bundle[header.value] = state[header.value];
+      }
+    });
+
+    axios.put(`/collection/${selectedCollection}?id=${bundle.id}`, bundle).then(() => {
+      queryClient.invalidateQueries('collections');
```

Comment:
Query keys can be more complex than a string. Instead of invalidating all collections queries it looks like the query key is set to `['collections', { selectedCollection }]` (Collections.jsx:83). Use the full key for all of these invalidateQueries so only the cache for that specific collection is invalidated.